test/dynamo/test_misc.py

```diff
@@ -9111,6 +9111,11 @@ def fn(img):
         self.assertEqual(msg, "shape torch.Size([8, 8]) batch size 1.00")
         self.assertEqual(res, img1 + torch.sin(img1))
 
+    # Compiling autograd.Function traces fwd function twice, but the same unbacked symints were not identified
+    # as the same across the two tracings. This is an unlikely situation in real use cases, so we add another
+    # `test_validate_outputs_unbacked_by_custom_op` to mitigate it and keep this one as expected failure
+    # until we have a proper fix.
```

Comment:
This was the problem that we had before, right? Are we just saying we don't care about it? There are alternative designs we can do to support this.

Alternative design proposal:
- run speculate subgraph
- do not do a separate FakeTensor pass
- instead, just do exactly what the autograd.Function logic in C++ does:
    - create a grad_fn and put it onto the FakeTensor. This grad_fn should probably be some dummy grad_fn. If this is too difficult, then I'm fine with a solution that just sets the output Tensor's requires_grad to be True.
    - Also, there is a special case for what happens if the autograd.Function returns the input directly. In that case we want to alias the input, detach, and then slam a grad_fn onto the Tensor.